database/migrations/Concerns/CanModifySettings.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace Database\Migrations\Concerns;
+
+use Closure;
+use Illuminate\Support\Facades\DB;
+
+trait CanModifySettings
+{
+    public function updateSettings(string $group, string $name, Closure $modifyPayload, bool $isEncrypted = false): void
```

Comment:
What are your thoughts on how we could possibly make the `modifyPayload` argument safer for inferring the arguments passed?

Though this is a really simple example with only one argument and not something we need to tackle immediately, I would like to figure out an alternative or a better way to type hinting `Closure`.

Though it works, it feels like we should either add information in docs or a docblock or requires that the consumer read through the code of `updateSettings` to understand what arguments are expected to be provided. There is no type safety or help in defining the arguments of the Closure when using the API.

I don't think there is a native way to define `Closure` arguments in docblocks (which still isn't perfect as it can become out of date)
My other thought is to use something like an Interface or Abstract class instead of `Closure` but that just sounds like a mess having to define a whole new class every time you want to call a method.